benchmark/simple_consumer_bench.py

```diff
@@ -0,0 +1,154 @@
+import subprocess
+import timeit
+
+from pykafka import KafkaClient
+from pykafka.test.utils import get_cluster
+
+
+NUM_PARTITIONS_RANGE = [4, 8, 16, 32, 64]
+MSG_SIZE_BYTES_RANGE = [100, 1000, 10**4, 10**5]
+
+
+def get_topic_name(num_partitions, msg_size_bytes):
+    """Generate test-topic names describing message characteristics"""
+    return "parts{}-size{}".format(num_partitions, msg_size_bytes)
+
+
+def mk_topics(num_partitions_range=NUM_PARTITIONS_RANGE,
+              msg_size_bytes_range=MSG_SIZE_BYTES_RANGE,
+              msg_sum_bytes=10**9,
+              max_num_msgs=10**6):
+    """
+    Pre-populate an existing testinstances cluster with test messages
+
+    A topic is created for each combination of partition-count and
+    message-size (passed in iterables).  Note that it's easy to eat lots of
+    disk space with this!
+
+    :param msg_sum_bytes: Limit the number of messages written to a topic
+                          so that topic will not exceed this number of bytes
+    :param max_num_msgs: Limit to number of messages written to any topic
+    """
+    cluster = get_cluster()  # make this in advance, like in travis.yml
+    topics = [(np, sz, get_topic_name(np, sz))
+              for np in num_partitions_range
+              for sz in msg_size_bytes_range]
+
+    for num_partitions, msg_size_bytes, topic_name in topics:
+        cluster.create_topic(topic_name, num_partitions, replication_factor=1)
+        print "Producing into {}".format(topic_name)
+        client = KafkaClient(cluster.brokers)
+        n_msgs = min(msg_sum_bytes // msg_size_bytes, max_num_msgs)
+        prod = client.topics[topic_name].get_producer()
+        prod.produce(msg_size_bytes * b" " for _ in xrange(n_msgs))
+
+
+SETUP = ("from pykafka import KafkaClient\n"
+         "from pykafka.rdkafka import RdKafkaSimpleConsumer\n"
+         "from pykafka.common import OffsetType\n"
+         "client = KafkaClient('localhost:9092')\n"
+         "topic = client.topics['{topic_name}']\n"
+         "kwargs = dict(\n"
+         "    auto_offset_reset=OffsetType.EARLIEST,\n"
+         "    consumer_timeout_ms=5000,\n"  # with default (-1) it might hang
+         "    num_consumer_fetchers={num_consumer_fetchers},\n"
+         "    queued_max_messages={queued_max_messages},\n"
+         "    )\n")
+SETUP_FIN = {
+    "pure-py": "cons = topic.get_simple_consumer(**kwargs)",
+    "rdkafka": "cons = RdKafkaSimpleConsumer(topic, client.cluster, **kwargs)",
+    }
+
+
+def run_bench(consumer_type,
+              num_partitions,
+              msg_size_bytes,
+              num_consumer_fetchers,
+              queued_max_messages,
+              num_iterations,
+              filename_append="consumer_bench.json"):
+    """
+    Run a single timeit-benchmark for num_iterations of consume() calls
+
+    This is a somewhat awkward-looking runner, because we currently have an
+    issue where consumers don't get garbage-collected, and so their worker
```

Comment:
No it's not. Actually the whole benchmark dir has gone very stale, and as we discussed over email back then, the methods could be better. Should I remove it all for now, and commit it when there's a fresh and more fair version?